wrapper/src/main/java/software/amazon/jdbc/plugin/bluegreen/BlueGreenStatusProvider.java

```diff
@@ -608,31 +617,43 @@ protected BlueGreenStatus getStatusOfPreparation() {
 
   protected List<ConnectRouting> addSubstituteBlueWithIpAddressConnectRouting() {
     List<ConnectRouting> connectRouting = new ArrayList<>();
-    this.roleByHost.entrySet().stream()
-        .filter(x -> x.getValue() == BlueGreenRole.SOURCE && this.correspondingNodes.containsKey(x.getKey()))
-        .forEach(x -> {
-          HostSpec hostSpec = this.correspondingNodes.get(x.getKey()).getValue1();
-          Optional<String> blueIp = this.hostIpAddresses.get(hostSpec.getHost());
-          HostSpec substituteHostSpecWithIp = blueIp == null || !blueIp.isPresent()
-              ? hostSpec
-              : this.hostSpecBuilder.copyFrom(hostSpec).host(blueIp.get()).build();
-
-          connectRouting.add(new SubstituteConnectRouting(
-              x.getKey(),
-              x.getValue(),
-              substituteHostSpecWithIp,
-              Collections.singletonList(hostSpec),
-              null));
-
-          connectRouting.add(new SubstituteConnectRouting(
-              this.getHostAndPort(
-                  x.getKey(),
-                  this.interimStatuses[x.getValue().getValue()].port),
-              x.getValue(),
-              substituteHostSpecWithIp,
-              Collections.singletonList(hostSpec),
-              null));
-        });
+    for (Map.Entry<String, BlueGreenRole> entry : this.roleByHost.entrySet()) {
+      String host = entry.getKey();
+      BlueGreenRole role = entry.getValue();
+      Pair<HostSpec, HostSpec> nodePair = this.correspondingNodes.get(host);
+      if (role == BlueGreenRole.TARGET || nodePair == null) {
```

Comment:
I'd suggest the following code as it's more readable:

`if (role != BlueGreenRole.SOURCE || nodePair == null) {`